src/workers/webdav/modules/files/application/WebdavFileCreator.ts

```diff
@@ -91,55 +96,84 @@ export class WebdavFileCreator {
     return file;
   }
 
-  async run(
-    path: string,
-    size: number
-  ): Promise<{
-    stream: Writable;
-    upload: Promise<WebdavFile['fileId']>;
-  }> {
-    const fileSize = new FileSize(size);
+  async run(path: string): Promise<{ stream: Writable }> {
     const filePath = new FilePath(path);
 
-    const metadata = ItemMetadata.from({
-      createdAt: Date.now(),
-      updatedAt: Date.now(),
-      name: filePath.name(),
-      size,
-      extension: filePath.extension(),
-      type: 'FILE',
+    const fileValidator = new WebdavFileValidator();
+
+    const isValid = fileValidator.validateName(filePath.nameWithExtension());
+
+    // Invalid files goes nowhere
+    if (!isValid) return { stream: new PassThrough() };
```

Comment:
Why don't we throw an error, if we return a stream, we will return "good" HTTP status, and the clients will think they are uploading a file to that writable

Reply:
yeah I think it will be better